app/services/migration/users/creator.rb

```diff
@@ -0,0 +1,119 @@
+module Migration
+  module Users
+    class Creator
+      attr_reader :ecf_user
+
+      def initialize(ecf_user)
+        @ecf_user = ecf_user
+      end
+
+      def find_or_initialize
+        if user_does_not_exist_on_npq?
+          return ::User.new(ecf_id: ecf_user.id)
+        end
+
+        raise_on_multiple_ecf_users_with_same_ecf_id!
```

Comment:
Should this go right at the top? otherwise we would miss those where the user doesn't exist in ECF potentially? 🤔